### PR TITLE
Migrated scheduler file `cache.go` to structured logging

### PR DESCRIPTION
**Pull request by shivanshuraj1333**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/issues/105841

#### Special notes for your reviewer:
Added missing migration points, fixed previously migrated logs
#### Does this PR introduce a user-facing change?

```release-note
Migrated scheduler file `cache.go` to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/cache/cache.go [hunk 1]
```diff
@@ -294,7 +295,7 @@ func (cache *schedulerCache) updateNodeInfoSnapshotList(snapshot *Snapshot, upda
 					snapshot.havePodsWithRequiredAntiAffinityNodeInfoList = append(snapshot.havePodsWithRequiredAntiAffinityNodeInfoList, nodeInfo)
 				}
 			} else {
-				klog.Errorf("node %q exist in nodeTree but not in NodeInfoMap, this should not happen.", nodeName)
+				klog.ErrorS(nil, "Node exist in nodeTree but not in NodeInfoMap, this should not happen", "node", klog.KRef("", nodeName))
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:298]: done, thanks :)

File: pkg/scheduler/internal/cache/cache.go [hunk 2]
```diff
@@ -381,7 +382,7 @@ func (cache *schedulerCache) finishBinding(pod *v1.Pod, now time.Time) error {
 	cache.mu.RLock()
 	defer cache.mu.RUnlock()
 
-	klog.V(5).Infof("Finished binding for pod %v. Can be expired.", key)
+	klog.V(5).InfoS("Finished binding for pod, can be expired", "key", key)
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:385]: done, thanks :)

File: pkg/scheduler/internal/cache/cache.go [hunk 3]
```diff
@@ -475,10 +476,10 @@ func (cache *schedulerCache) AddPod(pod *v1.Pod) error {
 	case ok && cache.assumedPods.Has(key):
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
 			// The pod was added to a different node than it was assumed to.
-			klog.Warningf("Pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
+			klog.InfoS("Pod was added to a different node than it was assumed", "key", key, "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:479]: done, thanks :)

File: pkg/scheduler/internal/cache/cache.go [hunk 4]
```diff
@@ -541,8 +543,9 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 	// before Remove event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods.Has(key):
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
-			klog.Errorf("Pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.InfoS("Pod was added to a different node than it was assumed", "key", key, "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:546]: done, thanks :)

File: pkg/scheduler/internal/cache/cache.go [hunk 5]
```diff
@@ -726,17 +729,18 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
...
 		ps, ok := cache.podStates[key]
 		if !ok {
-			klog.Fatal("Key found in assumed set but not in podStates. Potentially a logical error.")
+			klog.ErrorS(nil, "Key found in assumed set but not in podStates, potentially a logical error")
+			os.Exit(1)
 		}
 		if !ps.bindingFinished {
-			klog.V(5).Infof("Couldn't expire cache for pod %v/%v. Binding is still in progress.",
-				ps.pod.Namespace, ps.pod.Name)
+			klog.V(5).InfoS("Couldn't expire cache for pod, binding is still in progress",
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:736]: done, thanks :)

File: pkg/scheduler/internal/cache/cache.go [hunk 6]
```diff
@@ -513,8 +514,9 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods.Has(key):
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.ErrorS(nil, "Pod updated on a different node than previously added to", "key", key)
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:517]: should be `"pod", klog.KObj(oldPod)`, done thanks!

File: pkg/scheduler/internal/cache/cache.go [hunk 7]
```diff
@@ -726,17 +729,18 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
...
+			klog.V(5).InfoS("Couldn't expire cache for pod, binding is still in progress",
+				"pod", klog.KObj(ps.pod))
 			continue
 		}
 		if now.After(*ps.deadline) {
-			klog.Warningf("Pod %s/%s expired", ps.pod.Namespace, ps.pod.Name)
+			klog.InfoS("Pod expired", "pod", klog.KObj(ps.pod))
 			if err := cache.expirePod(key, ps); err != nil {
-				klog.Errorf("ExpirePod failed for %s: %v", key, err)
+				klog.ErrorS(err, "ExpirePod failed", "key", key)
```
[the remark this answers is not in the document]
Comment (shivanshu1333) [reply pkg/scheduler/internal/cache/cache.go:743]: should be `"pod", klog.KObj(ps.pod)`, done thanks!

**Comment by shivanshu1333** [comment]
ping @Huang-Wei for approval :)

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/assign @damemi

For approval.

**Comment by serathius** [comment]
This PR is needed to finish scheduler structured logging migration planned for 1.23
/milestone v1.23

**Comment by shivanshu1333** [comment]
ping @damemi for approval

**Pull request merged**